Examples/test-suite/typescript_optional.i

```diff
@@ -0,0 +1,19 @@
+%module typescript_optional
+
+%typemap (typescriptoptional) OptionalD "";
```

Comment:
It seems really strange to define a typemap as an empty string. I think it would be better to require it be defined as "1". This would still be unusual (I don't think there are any other typemaps like this), but IMO a bit less so.

BTW, thinking more about this, a more SWIGgish way of doing this could be to add an argument to `typescripttype` typemap, i.e. write this as
```
%typemap(typescripttype, optional="1") OptionalID "number"
```
There are examples of such attributes in the typemap library already, e.g. `nopgcpp` for C# and `nouse` for Java.